### DISCUSSION TITLE
WeatherFlow v1.0.2 not showing in HA integration list

### DISCUSSION THREAD

**Discussion by menaibluey** [briis/hass-weatherflow#27] [General]
I'm running latest HA version 2021.12.9. Download WeatherFlow Integration v1.0.2 via HACS. When trying to implement this via "Add Integration"  in Configuration the WeatherFlow component does not show in the list of integrations. I have double-checked that the folder and files as per the manual installation instructions are present. I have also restarted HA and even completely restarted the server. I'm running HA as a virtual machine on my Synology Diskstation. There has never been a problem with the previous WeatherFlow integration.

Is there anything else I need to do so it shows up in the list of available integrations?

**Comment by juanscybaski** [comment]
I am having the same problem.

**Comment by briis** [comment]
In 99% of all cases this is a caching issue with your Web Browser.

Try this:
- Go to the *Integrations Page* and when on this page do **Hard Refresh** of your Browser (On a Mac this is normally CMD+R)
- Now try and add the integration again.

If the above does not work, I found that there has been a Bug in HACS that placed Integrations in the wrong directory. It was introduced in V1.19 of HACS and is most likely fixed in 1.19.2. To check if the Integration is stored correctly, check the directory `custom_components/weatherflow` this directory should NOT contain a new subdirectory called `weatherflow`. If it does, move all the files from this directory to `custom_components/weatherflow` and restart HA. Then you should be able to install the Integration.

And remember to update HACS to the latest version.
